lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -304,12 +304,13 @@ def merge_credit_card_options(parameters, options)
 
         parameters[:credit_card] ||= {}
         parameters[:credit_card].merge!(:options => valid_options)
-        parameters[:credit_card][:billing_address] = map_address(options[:billing_address]) if options[:billing_address]
+        address = options[:billing_address]&.except(:phone)
+        return parameters if address.nil? || address.empty?
```

Comment:
This can be `address&.empty?` but I don't really think that's worth altering the patch.

Reply:
Actually we need to be checking for either case. `&.empty?` lets `nil` through.

Reply:
`nil` and `false` are bothy falsy, and `nil&.foo` return `nil`, so I think it would indeed be fine to be `... if address&.empty?`. E.g.,

```
irb(main):001:0> puts 'hello' if nil&.empty?
=> nil
irb(main):002:0> 
```

But again, I'm not convinced the half line deletion is terribly useful here.

Reply:
Interesting, because how I described it is how it seemed to behave in testing. I'll experiment some more for the sake of understanding.

Reply:
```
irb(main):006:0> a = nil
=> nil
irb(main):007:0> puts "asdf" if a&.empty?
=> nil
```

Reply:
So the thing is, in your example, we want 'hello', but it's returning nil instead.

Reply:
Bah, you're totally right; `||`, not `&&`. I got doubly turned around here because `nil.empty?` is defined (and returns `true`) if Active Support is loaded.

Reply:
I think the confusion lies here:
```

Reply:
Aaah, ok gotcha. Nevermind the above 👍